src/langcheck/metrics/custom_text_quality.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from jinja2 import Environment, Template, meta
+
+from langcheck.metrics._validation import validate_parameters_custom_evaluator
+from langcheck.metrics.eval_clients import EvalClient
+from langcheck.metrics.metric_value import MetricValue
+
+
+def custom_evaluator(generated_outputs: list[str] | str | None,
+                     prompts: list[str] | str | None,
+                     sources: list[str] | str | None,
+                     reference_outputs: list[str] | str | None,
+                     eval_model: EvalClient, metric_name: str,
+                     score_map: dict[str, float], template_path: str,
+                     language: str) -> MetricValue[float | None]:
+    '''Calculates the scores of a custom evaluator. The EvalClient will first
+    assess the provided inputs using the prompt template, and then convert those
+    assessments into scores using the score map.
+
+    Args:
+        generated_outputs: The model generated output(s)
+        prompts: The prompts used to generate the output(s)
+        sources: The source(s) of the generated output(s)
+        reference_outputs: The reference output(s)
+        eval_model: The EvalClient instance used for the evaluation
+        metric_name: The name of the metric
+        score_map: A dictionary mapping the evaluator's assessments to scores
+        template_path: The path to the prompt template file
+        language: The language that the evaluator will use ('en', 'ja', or 'de')
+
+    Returns:
+        A MetricValue object
+    '''
+    generated_outputs, prompts, reference_outputs, sources = validate_parameters_custom_evaluator(  # NOQA: E501
+        generated_outputs, prompts, reference_outputs, sources)
+    # Find the length of the first non-None list (they are guaranteed to all be
+    # the same length)
+    num_examples = next(
```

Comment:
No a strong opinion, but maybe we can do that in the `validate_parameters_` function?

Reply:
You mean that we should return `num_examples` from `validate_parameters_custom_evaluator`?

Reply:
Yes!
But now I think it could mess up the return value type, let's leave it as it is